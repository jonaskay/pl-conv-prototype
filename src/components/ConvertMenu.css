.ConvertMenu {
  box-shadow:
    0 1px 5px 0 rgba(0,0,0,.2), 0 2px 2px 0 rgba(0,0,0,.14), 0 3px 1px -2px rgba(0,0,0,.12);
  box-sizing: border-box;
  margin-bottom: 1.5rem;
  width: 100%;
}

.ConvertMenu-header {
  background: #424242;
  font-size: 1.25rem;
  padding: 1rem;
}

.ConvertMenu-body {
  background: #303030;
  padding: 1rem;
}
