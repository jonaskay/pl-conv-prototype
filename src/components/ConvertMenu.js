import React from 'react';
import './ConvertMenu.css';
import ConvertMenuField from './ConvertMenuField';

class ConvertMenu extends React.Component {
  getSpotifyAuthorizeUrl() {
    let url = 'https://accounts.spotify.com/authorize?'; 
    url += 'client_id=' + process.env.REACT_APP_SPOTIFY_CLIENT_ID;
    url += '&redirect_uri=' + process.env.REACT_APP_SPOTIFY_REDIRECT_URI;
    url += '&scope=playlist-modify-private&response_type=token';
    return url
  }

  handleSpotifyButtonClick = () => {
    window.location = this.getSpotifyAuthorizeUrl();
  }

  render() {
    return (
      <div className="ConvertMenu">
        <div className="ConvertMenu-header">Settings</div>
        <div className="ConvertMenu-body">
          <div style={{color: 'rgba(255,255,255,.7)'}}>Convert playlist to:</div>
          <ConvertMenuField 
            name="Spotify" 
            authenticated={this.props.spotifyAuthenticated}
            checked={this.props.convertToSpotify}
            onInputClick={this.props.onSpotifyToggle}
            onButtonClick={this.handleSpotifyButtonClick} />
          <ConvertMenuField name="YouTube" authDisabled={true} />
        </div>
      </div>
    );
  }
}

export default ConvertMenu;
