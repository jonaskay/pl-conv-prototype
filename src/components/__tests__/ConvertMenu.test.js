import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ConvertMenu from '../ConvertMenu';

Enzyme.configure({ adapter: new Adapter() });

it('renders a menu', () => {
  const component = shallow(<ConvertMenu />);
  expect(component).toMatchSnapshot();
});

describe('getSpotifyAuthorizeUrl', () => {
  const component = shallow(<ConvertMenu />);
  expect(component.instance().getSpotifyAuthorizeUrl()).toMatchSnapshot();
});
