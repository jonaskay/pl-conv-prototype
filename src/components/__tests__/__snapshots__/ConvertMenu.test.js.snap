// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"https://accounts.spotify.com/authorize?client_id=undefined&redirect_uri=undefined&scope=playlist-modify-private&response_type=token"`;

exports[`renders a menu 1`] = `
<div
  className="ConvertMenu"
>
  <div
    className="ConvertMenu-header"
  >
    Settings
  </div>
  <div
    className="ConvertMenu-body"
  >
    <div
      style={
        Object {
          "color": "rgba(255,255,255,.7)",
        }
      }
    >
      Convert playlist to:
    </div>
    <ConvertMenuField
      name="Spotify"
      onButtonClick={[Function]}
    />
    <ConvertMenuField
      authDisabled={true}
      name="YouTube"
    />
  </div>
</div>
`;
